internal/pkg/util/env/env.go

```diff
@@ -74,3 +79,11 @@ func GetenvLegacy(key, legacyKey string) string {
 func TrimApptainerKey(key string) string {
 	return strings.TrimPrefix(key, ApptainerPrefixes[0])
 }
+
+func DefaultLocalKeyDirPath() string {
+	// read this as: look for APPTAINER_KEYSDIR and/or SINGULARITY_SYPGPDIR
+	if dir := GetenvLegacy("KEYSDIR", "SYPGPDIR"); dir != "" {
```

Comment:
@JasonYangShadow What are your thoughts on taking this environment variable parsing logic and placing it into the `EnvHandler` of the `--keysdir` flag? https://github.com/apptainer/apptainer/blob/main/pkg/cmdline/flag.go#L38

Reply:
@ikaneshiro there is one line here
https://github.com/JasonYangShadow/apptainer/blob/1cea4bf352df81e316f78641be52e4e2582fbc8e/pkg/sypgp/sypgp.go#L122
also calls the function.  😅

Reply:
as we discussed, I will split this function into two parts, one for EnvHandler, and another one for the fallback in NewHandle, I will refactor the changes tomorrow. Thanks for the really good suggestion.

Reply:
@ikaneshiro I tried customized EnvHandler with the new flag, but it looks like the handler is not called. Not sure about the reason and I am still working on it.

Reply:
@JasonYangShadow It looks like `EnvHander` doesn't quite provide the functionality the functionality I hoped it would. Thanks for testing, sorry for the false lead here.

Reply:
@ikaneshiro ohh, no problem at all. Let's figure out the root cause and maybe we can improve its functionality.